### PR TITLE
PSIMA-132: Updating Thermal  struct

### PR DESCRIPTION
**Pull request by sourabhdalvi**
Adding `timeatstatus` field for initial conditions in ThermalStandard

### CODE REVIEW

File: src/models/generated/ThermalStandard.jl [hunk 1]
```diff
@@ -77,12 +80,12 @@ mutable struct ThermalStandard <: ThermalGen
...
+    ThermalStandard(name, available, status, timeatstatus, bus, activepower, reactivepower, rating, primemover, fuel, activepowerlimits, reactivepowerlimits, ramplimits, timelimits, op_cost, services, participation_factor, ext, forecasts, InfrastructureSystemsInternal(), )
 end
 
-function ThermalStandard(; name, available, status, bus, activepower, reactivepower, rating, primemover, fuel, activepowerlimits, reactivepowerlimits, ramplimits, timelimits, op_cost, services=Device[], participation_factor=0.0, ext=Dict{String, Any}(), forecasts=InfrastructureSystems.Forecasts(), )
-    ThermalStandard(name, available, status, bus, activepower, reactivepower, rating, primemover, fuel, activepowerlimits, reactivepowerlimits, ramplimits, timelimits, op_cost, services, participation_factor, ext, forecasts, )
+function ThermalStandard(; name, available, status, timeatstatus, bus, activepower, reactivepower, rating, primemover, fuel, activepowerlimits, reactivepowerlimits, ramplimits, timelimits, op_cost, services=Device[], participation_factor=0.0, ext=Dict{String, Any}(), forecasts=InfrastructureSystems.Forecasts(), )
```
Comment (jd-lara) [inline src/models/generated/ThermalStandard.jl:87]: There should be a default value here. Follow the example of the participation factor

File: src/descriptors/power_system_structs.json [hunk 2]
```diff
@@ -1678,6 +1678,11 @@
                 "name": "status",
                 "data_type": "Bool"
             },
+            {
+                "null_value": "Inf",
+                "name": "timeatstatus",
+                "data_type": "Float64"
```
Comment (jd-lara) [inline src/descriptors/power_system_structs.json:1684]:
```suggestion
                "data_type": "Float64"
                "default": "Inf",
```

File: src/descriptors/power_system_structs.json [hunk 3]
```diff
@@ -1678,6 +1678,11 @@
                 "name": "status",
                 "data_type": "Bool"
             },
+            {
+                "null_value": "Inf",
+                "name": "timeatstatus",
```
Comment (daniel-thom) [inline src/descriptors/power_system_structs.json:1683]: time_at_status

File: src/common.jl [hunk 4]
```diff
@@ -79,3 +79,5 @@ const POWER_SYSTEM_STRUCT_DESCRIPTOR_FILE =
     joinpath(dirname(pathof(PowerSystems)), "descriptors", "power_system_structs.json")
 
 const DEFAULT_SYSTEM_FREQUENCY = 60.0
+
+const INFINITY = 1e6
```
Comment (jd-lara) [inline src/common.jl:83]:
```suggestion
const INFINITE_TIME = 1e6
```

**Pull request merged**